express-mutations/server.ts

```diff
@@ -0,0 +1,121 @@
+import express from 'express';
+import pg from 'pg';
+import { ClientError, errorMiddleware } from './lib/index.js';
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/pagila',
+  ssl: {
+    // Allow non-SSL traffic to localhost
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+
+app.use(express.json());
+
+app.post('/api/actors', async (req, res, next) => {
+  try {
+    const { firstName, lastName } = req.body;
+    if (firstName === undefined) {
+      throw new ClientError(400, 'firstName is undefined');
+    }
+    if (lastName === undefined) {
+      throw new ClientError(400, 'lastName is undefined');
+    }
+
+    const sql = `
+    insert into "actors" ("firstName", "lastName")
+    values ($1, $2)
+    returning *
+    `;
+    const params = [firstName, lastName];
+    const results = await db.query(sql, params);
+    const post = results.rows[0];
+
+    res.status(201).json(post);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.put('/api/actors/:actorId', async (req, res, next) => {
+  try {
+    const { firstName, lastName } = req.body;
+    if (lastName === undefined) {
+      throw new ClientError(400, 'lastName is undefined');
+    } else if (firstName === undefined) {
+      throw new ClientError(400, 'firstName is undefined');
+    }
+    const { actorId } = req.params;
+    if (actorId === undefined) {
+      throw new ClientError(400, 'ActorId is undefined');
+    }
```

Comment:
You should also check that the actorId is an integer. Otherwise you'll get a SQL exception.